src/app/drive/views/DriveView/DriveView.tsx

```diff
@@ -81,6 +85,14 @@ const DriveView = (props: DriveViewProps) => {
     }
   };
 
+  const setPersonalWithUrl = (itemUuid) => {
+    dispatch(workspacesActions.setCredentials(null));
+    dispatch(workspacesActions.setSelectedWorkspace(null));
+    localStorageService.set(STORAGE_KEYS.WORKSPACE_CREDENTIALS, 'null');
+    localStorageService.set(STORAGE_KEYS.B2B_WORKSPACE, 'null');
+    isFolderView ? goFolder(itemUuid) : showFile(itemUuid);
```

Comment:
I think it will be better to keep the navigation function out of this function that sets the workspace so that the function focuses the behaviour on one thing only

Reply:
Done